test/README.md

```diff
@@ -0,0 +1,166 @@
+# Cloud Operations Puppet Agents Testing
+
+## Overview
+
+This repo leverages the following for integration testing:
+
+- [Google Compute Engine](https://cloud.google.com/compute): Test instances
+- [Github Actions](https://github.com/features/actions): Continuous integration
+- [Packer](https://www.packer.io/): Build test instance base images
+- [Terraform](https://www.terraform.io/): Deploy test instances
+- [Inspec](https://github.com/inspec/inspec): Perform validation
+
+## Google Compute Engine
+
+Google Compute Engine hosts the testing environment, ensure the following APIs are enabled in the project:
+
+```bash
+gcloud services enable compute.googleapis.com
+gcloud services enable storage-api.googleapis.com
+```
+
+Create a service account with the following roles:
+
+- Compute Admin
+- Service Account User
+
+Create a service account json key, which will be referenced during Github Actions configuration.
+
+## Github Actions
+
+Github actions requires several secrets. Navigate to the repositories secrets and add the following:
+
+- **GCP_PROJECT_ID**: Google Cloud project_id
+- **GCP_SA_KEY**: Google Cloud service account json key
+- **SSH_PRIVATE_KEY**: keypair private key
+- **SSH_PUBLIC_KEY**: keypair public key
+
+#### Keypair
+
+You can generate an SSH keypair with `ssh-keygen`. This keypair will be used for Puppet and Inspec's ssh connectivity.
+
+## Packer
+
+Packer will require access to Google APIs. This [guide provides](https://cloud.google.com/build/docs/building/build-vm-images-with-packer) examples on how to setup your working environment.
+
+### Linux
```

Comment:
Are these images built once and hosted somewhere, or built from scratch every time during a test run? I guess for Windows it is probably hosted somewhere, since it's built manually, right?

Reply:
I pushed a commit clarifying this. Basically, Packer is used to prepare base images for Linux. The same process is done manually for Windows. Terraform will deploy these "baked" images for Linux and Windows tests every time CI runs. The Windows instances are not left behind to be re-used, they are created on demand.

The project being used for CI should own these images, or have read access to them if they are in a different project.